### PR TITLE
Add ability to use prepackaged configuration file.

### PR DESCRIPTION
**Pull request by RebeccaRGBA**

### CODE REVIEW

**Comment by flagxor** [comment]
Alon, we're playing around with cooking a .deb, this would help keep it from clashing with local user configs.

File: tools/shared.py [hunk 1]
```diff
@@ -207,7 +207,9 @@ def new(*args):
     EM_CONFIG = EM_CONFIG.replace(';', '\n') + '\n'
 
 if not EM_CONFIG:
-  EM_CONFIG = '~/.emscripten'
+  EM_CONFIG = path_from_root('emscripten.rc')
+  if not os.path.exists(EM_CONFIG):
+    EM_CONFIG = '~/.emscripten'
```
Comment (s3ththompson) [inline tools/shared.py]: In the future, perhaps it would be nice if the user config could inherit from the root config (if it exists).

**Comment by flagxor** [comment]
We discussed this after lunch, when it occurred to Rebecca that if the user has a local config file under the standard name, it will almost certainly override the system one (defeating the purpose, i.e. to allow the package to be hermetic by default).
We concluded that a package trying to alter the behavior around this would probably want to select an alternate default config file name, and do the import itself.

**Comment by kripken** [comment]
I think I don't fully understand the use case here. We already support setting the location of the config file using either an env var or a commandline parameter. Is this adding another way to do it, by looking for a file in the current directory? If that's useful in some cases it sounds like a good idea, I just don't see the full picture here yet I think.

**Comment by juj** [comment]
It sounds like the scenario would be that the installation would be prohibited from adding an env. var, afaik Linuxes or Mac OS X don't generally support a scheme for installer-loaded env. vars like Windows does? Also, injecting an appropriate `EM_CONFIG` parameter to command lines of each client build invocation would not be feasible.

However my question is that how is the packaging intended to work if `.emscripten` lives inside the Emscripten root directory? How does it know where to point to to find the fastcomp LLVM build? Or is the intent to copy fastcomp LLVM binaries over to the same Emscripten directory tree as well so that they can be found in a relative path?

**Comment by kripken** [comment]
This PR is quite old, closing. Feel free to reopen if still relevant.

**Pull request closed without merging**